### PR TITLE
feat: Add tag list and feed posts of followings

### PR DESCRIPTION
**Pull request by softwareVirus**
## Features Added
- **Tag Lists:**
  - Implemented tag functionality allowing users to add tags to their posts.
  - Added endpoints for managing tags associated with user posts.
  - Users can now filter posts based on tags.

- **Feed Posts:**
  - Implemented feed posts functionality to replace "My Posts" with posts from followed users.
  - Added routes and methods to retrieve feed posts based on users' following list.
  - Users can now view posts from users they are following.

## Changes Made
- Modified the post resource to handle tag-related operations and feed post retrieval.
- Updated the user model to include fields for tags, followers, and following.
- Adjusted the client-side components to accommodate tag lists and display feed posts.

### CODE REVIEW

File: client/src/components/UserProfile/SfUserProfile.vue [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        userId() {
+            return this.$route.params.id
+        },
+        isFollowing() {
+            return this.$store.state.user?.following.includes(this.userId) === true
+        }
+    },
+    watch: {
+        async selectedTags(newValue, oldValue) {
+            console.log(newValue, oldValue)
```
Comment (aahmetdurmaz) [inline client/src/components/UserProfile/SfUserProfile.vue]: I see lot of console.log like this.

File: server/app/__init__.py [hunk 2]
```diff
@@ -94,24 +96,37 @@ def user_lookup_loader(_jwt_header, jwt_data):
...
 api.add_resource(LogoutResource, "/logout")
 api.add_resource(RefreshResource, "/refresh_token")
+api.add_resource(TagResource, "/tags", "/tags/<string:user_id>")
+
 
 def create_app(mode="dev"):
-    #with open(f"./config.json", "r") as f:
+    # with open(f"./config.json", "r") as f:
     #    config = json.load(f)
     print(mode)
```
Comment (aahmetdurmaz) [inline server/app/__init__.py:120]: I don't know if it's deliberately, but i think u don't need this print.

File: server/app/__init__.py [hunk 3]
```diff
@@ -94,24 +96,37 @@ def user_lookup_loader(_jwt_header, jwt_data):
...
     if mode == "dev":
-        connect(host="mongodb+srv://hello:12053085408a@cluster0.pn5ujkz.mongodb.net/?retryWrites=true&w=majority")#host=config["dev"]["MONGODB_URI"])
+        connect(
+            host="mongodb+srv://hello:12053085408a@cluster0.pn5ujkz.mongodb.net/?retryWrites=true&w=majority"
```
Comment (gecicidegisken) [inline server/app/__init__.py]: We need to avoid hard-coded sensitive information if possible

File: server/app/models/user.py [hunk 4]
```diff
@@ -42,6 +56,9 @@ class User(Document):
     last_name = StringField(max_length=200, required=True)
     email = EmailField(unique=True, required=True)
     hashed_password = StringField(required=True)
+    tags = ListField(ReferenceField("Tag"))
```
Comment (gecicidegisken) [inline server/app/models/user.py:59]: Now I know how [it](https://github.com/softwareVirus/micro-blog/pull/24#discussion_r1538457237) works :D 
But I'm not sure if this is the best way of delivering these related changes

**Comment by dilekyalcin** [comment]
We should avoid sensitive information and console.logs should remove

File: client/src/components/UserProfile/SfUserProfile.vue [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+                tags: newValue
+            })
+        }
+    },
+    async mounted() {
+        this.profile = await this.getProfile(this.userId)
+        await this.getUserPosts({
+            userId: this.userId
+        })
+        console.log(this.profile, this.tags)
```
Comment (hilaldedek) [inline client/src/components/UserProfile/SfUserProfile.vue]: console.logs forgotten

File: client/src/store/index.js [hunk 6]
```diff
@@ -136,6 +140,26 @@ export const store = new Vuex.Store({
...
+        console.log(error);
+      }
+    },
+    getTags: async function ({commit}, user_id) {
+      try {
+        const response = await axiosApiInstance.get(`/tags${user_id ? '/'+user_id: ''}`);
+        commit('setTags', response.data)
+        console.log(response);
+      } catch (error) {
+        console.log(error);
```
Comment (hilaldedek) [inline client/src/store/index.js]: console.logs forgotten otherwise LGTM

**Pull request merged**